src/main/java/graphql/schema/idl/SchemaPrinter.java

```diff
@@ -635,13 +635,15 @@ private static String printAst(Object value, GraphQLInputType type) {
 
     private TypePrinter<GraphQLSchema> schemaPrinter() {
         return (out, schema, visibility) -> {
+            List<GraphQLDirective> schemaDirectives = schema.getSchemaDirectives();
             GraphQLObjectType queryType = schema.getQueryType();
             GraphQLObjectType mutationType = schema.getMutationType();
             GraphQLObjectType subscriptionType = schema.getSubscriptionType();
 
             // when serializing a GraphQL schema using the type system language, a
             // schema definition should be omitted if only uses the default root type names.
-            boolean needsSchemaPrinted = options.isIncludeSchemaDefinition();
+            boolean needsSchemaPrinted = options.isIncludeSchemaDefinition() ||
+                    (this.options.includeDirectiveDefinitions && !schemaDirectives.isEmpty());
```

Comment:
Actually this flag is wrong

includeDirectiveDefinitions controls printing the actual directive definitions - not the uses of them.

Also if isIncludeSchemaDefinition was false BUT it had directives it would print.

We dont want this.  I think you should leave this as it was.

There is a `Predicate<GraphQLDirective> includeDirective;` in the options that can be used to turn off printing of directives

Reply:
Please put this back to how it was

Reply:
Thanks for pointing this out. I'll put this back to how it was.